### DISCUSSION TITLE
Is there any way to pin a task to the special CPU core?

### DISCUSSION THREAD

**Discussion by jeffer80** [nanoframework/Home#1291] [Q&A] [answered]
I want to run a high-priority task by using CPU1, and others using CPU0
In esp-idf the function xTaskCreatePinnedToCore can pin the task to a designated core,
and is there any way that I can do like that in nanoframework?

https://docs.espressif.com/projects/esp-idf/zh_CN/v5.0.1/esp32/api-reference/system/freertos.html#_CPPv423xTaskCreatePinnedToCore14TaskFunction_tPCKcK8uint32_tPCv11UBaseType_tPC12TaskHandle_tK10BaseType_t

**Comment by alberk8** [comment, marked as answer]
The nanoCLR is pinned to Core1 of ESP32 and the Wire Protocol is pinned to Core0.   C# program do not have access to the underlying CPU cores.

**Comment by josesimoes** [comment]
To control priority of .NET Threads you have [ThreadPriority](https://docs.nanoframework.net/api/System.Threading.ThreadPriority.html).
